app/pages/Reserved/index.js

```diff
@@ -0,0 +1,379 @@
+      }
+    }
+    if (!found) {
+      this.setState({
+        error: `Can not sign with key ${fileName}`,
+      });
+      return;
+    }
+
+    const keyFile = fs.readFileSync(filePath, 'utf-8');
+    const [alg, privateKey] = dnssec.decodePrivate(keyFile);
+
+    const rr = this.getTXT();
+    const lifespan = 365 * 24 * 60 * 60;
+
+    const sig = dnssec.sign(this.state.keys[0], privateKey, [rr], lifespan);
+
+    const proof = this.state.proof;
+    const zone = proof.zones[proof.zones.length - 1];
+    zone.claim.push(rr);
+    zone.claim.push(sig);
+
+    let json;
+    try {
+      json = await node.sendRawClaim(proof.toBase64());
+    } catch (e) {
+      this.setState({
+        error: `Error sending claim: ${e.message}`,
+      });
+      return;
+    }
+
+    // TODO: success modal with details from json
+  }
+
+  displayKeys() {
+    const rows = [];
+    for (let key of this.state.keys) {
+      const filename = key.filename;
+      key = key.getJSON()
+      rows.push(
+        <tr>
+          <td>{key.data.keyTag}</td>
+          <td>{key.data.keyType}</td>
+          <td>{key.data.algName}</td>
+          <td>{filename}</td>
+        </tr>
+      );
+    }
+
+    return (
+      <div>
+        &nbsp;&nbsp;DNSKEYs available for signing:
+        <table className="reserved__keyList">
+          <tr><td>Key tag</td><td>Key type</td><td>Key Algorithm</td><td>Filename</td></tr>
+          {rows}
+        </table>
+      </div>
+    );
+  }
+
+  render() {
+    const fakesig =
+      this.state.url ?
+        `${this.state.url}   300     IN      RRSIG   TXT 13 2 300 20210603180907 20210601160907 34505 ${this.state.url} f6x5CBP1ySenfPodSGSPNPCdzLzhlXK8shtpfzcEmCs09amCSqCIwniq eEIR1EYCuijP4OCKFyEnEhfEk+l81A==` :
+        '';
+
+    return (
+      <div>
+        <div className="reserved__top">
+          Search for reserved URL:&nbsp;
+          <input
+            type="text"
+            value={this.state.url}
+            onChange={this.handleInputValueChange}
+            onKeyDown={e => e.key === 'Enter' && this.searchReserved()}
+            placeholder="icann.org"
+          />
+          <h2>Claim reserved name: {this.state.name}</h2>
+        </div>
+        <div className="reserved__error">
+          {this.state.error}
+        </div>
+        <div>
+          <table className="reserved__table">
+            <tr><td>Reward</td><td>{this.state.claim.reward}</td><td></td></tr>
+            <tr><td>Fee</td><td>{this.state.claim.fee}</td><td></td></tr>
+            <tr><td>TXT</td><td className="wrap">{this.state.claim.txt}</td><td><CopyButton content={this.state.claim.txt} /></td></tr>
+          </table>
+        </div>
+        <div>
+          <h2>Sign and broadcast this claim:</h2>
+
+          <div className="reserved__method">
+            Method 1 (online): Paste above TXT into legacy DNS.
+            <Tooltipable
+              tooltipContent="TODO: blah blah blah DNSSEC, GoDaddy, then come back here and click...">
```

Comment:
![image](https://user-images.githubusercontent.com/1330573/120541657-6f2e4900-c3ea-11eb-96be-d3da4b662102.png)
🤣

Reply:
yeah... hence "draft" mode :-P

Reply:
looks alright to me

Reply:
Ok I filled in all the `TODO` tooltips... ;-)